pygeoapi_plugins/provider/geopandas_provider.py

```diff
@@ -0,0 +1,492 @@
+                    self.geometry_col = col
+                    self._exclude_from_fields.append(self.geometry_col)
+                    break
+            else:
+                # Assuming that there is no reason to read in geometric data without geometry
+                LOGGER.warning("No geometry column found")
+
+    def __init__(self, provider_def: dict):
+        """
+        Initialize object
+
+        :param provider_def: provider definition
+
+        :returns: pygeoapi.provider.base.GeoPandasProvider
+        """
+
+        super().__init__(provider_def)
+
+        try:
+            self.gdf = geopandas.read_file(provider_def['data'])
+        except FileNotFoundError as ex:
+            raise ProviderNoDataError(f"Tried to read GeoDataFrame: {ex} but it does not exist")
+        except Exception as ex:
+            raise ProviderInvalidDataError(f"Failed to read GeoDataFrame: {ex}")
+            
+             
+        # These fields should not be returned in the property list for a query
+        self._exclude_from_fields: list[str] = []
+
+        self._set_time_field(provider_def)
+        self._set_geometry_fields(provider_def)
+            
+        self.gdf[self.id_field] = self.gdf[self.id_field].astype(str)
+        
+        # Without below, the CSV reads std_id as an object dtype
+        # And fails the CSV provider tests. Maybe a way to do this better
+        # that is more generalizable?
+        if "stn_id" in self.gdf.columns:
+            self.gdf["stn_id"] = self.gdf["stn_id"].astype('int64')
+        if "value" in self.gdf.columns:
+            self.gdf["value"] = self.gdf["value"].astype('float64')
+
+        self._exclude_from_properties: list[str] = self._exclude_from_fields + [self.id_field]
+
+        self.fields = self.get_fields()
+
+        
+    def get_fields(self) -> dict[str, any]:
+        """
+        Get provider field information (names, types)
+
+        Example response: {'field1': 'string', 'field2': 'number'}}
+
+        :returns: dict of field names and their associated JSON Schema types
+        """
+        if len(self.gdf) == 0:
+            raise ProviderNoDataError("No data found to get fields from")
+        
+
+        field_mapper =  {col: self.gdf[col].dtype.name for col in self.gdf.columns 
+                         if col not in self._exclude_from_fields
+                         }
+        
+       # Pandas has a different ãmes for types than the pygeoapi spec expects
+        pandas_dtypes_to_ours = {
+            'float64': 'number',
+            'int64': 'integer',
+            'object': 'string'
+        }
+
+        pandas_default = defaultdict(lambda: 'string')
+        pandas_default.update(pandas_dtypes_to_ours)
+
+        
+        our_types_names = {k: {
+            "type": pandas_default[v]
+            }
+        for k, v in field_mapper.items()
+        }
+
+        return our_types_names
+
+    def get_schema(self, schema_type: SchemaType = SchemaType.item):
```

Comment:
No need to implement unimplemented methods of the Base Provider. For all of the methods returning `NotImplementedEror()`s